calcom/cal.com
# feat(web): improve session retrieval performance

## What does this PR do?

Switch to using `getServerSession` which avoids a HTTP round trip to retrieve session details.

Additionally, migrate deprecated `app/lib/auth` calls to to `@calcom/lib` package.

Fixes #7242 

**Environment**: Staging(main branch) / Production

## Type of change


## How should this be tested?

Run through critical flows and ensure nothing is broken, verify that pages and API endpoints are snappier.

See below table for speedup context:

| Path                                            | Before Avg | After Avg | Before Med | After Med | Before Min | After Min | Before Max | After Max |
| ----------------------------------------------- | ---------- | --------- | ---------- | --------- | ---------- | --------- | ---------- | --------- |
| /api/trpc/viewer.appById                        | 341.47ms   | 124.25ms  | 460.00ms   | 56.00ms   | 40.00ms    | 40.00ms   | 771.00ms   | 538.00ms  |
| /api/trpc/viewer.apps                           | 397.00ms   | 221.00ms  | 387.00ms   | 387.00ms  | 55.00ms    | 55.00ms   | 820.00ms   | 387.00ms  |
| /api/trpc/viewer.availability.list              | 299.00ms   | 171.50ms  | 295.00ms   | 295.00ms  | 48.00ms    | 48.00ms   | 613.00ms   | 295.00ms  |
| /api/trpc/viewer.availability.schedule.get      | 149.67ms   | 91.67ms   | 197.00ms   | 112.00ms  | 47.00ms    | 47.00ms   | 227.00ms   | 116.00ms  |
| /api/trpc/viewer.bookings.get                   | 276.00ms   | 300.00ms  | 300.00ms   | 300.00ms  | 252.00ms   | 300.00ms  | 300.00ms   | 300.00ms  |
| /api/trpc/viewer.bookingUnconfirmedCount        | 455.06ms   | 228.44ms  | 499.00ms   | 240.00ms  | 87.00ms    | 87.00ms   | 911.00ms   | 499.00ms  |
| /api/trpc/viewer.connectedCalendars             | 286.00ms   | 136.50ms  | 228.00ms   | 228.00ms  | 45.00ms    | 45.00ms   | 738.00ms   | 228.00ms  |
| /api/trpc/viewer.eventTypes.get                 | 151.50ms   | 65.00ms   | 238.00ms   | 65.00ms   | 65.00ms    | 65.00ms   | 238.00ms   | 65.00ms   |
| /api/trpc/viewer.eventTypes.getByViewer         | 343.24ms   | 218.00ms  | 260.00ms   | 209.00ms  | 157.00ms   | 157.00ms  | 783.00ms   | 351.00ms  |
| /api/trpc/viewer.getUsersDefaultConferencingApp | 67.50ms    | 15.00ms   | 120.00ms   | 15.00ms   | 15.00ms    | 15.00ms   | 120.00ms   | 15.00ms   |
| /api/trpc/viewer.integrations                   | 103.69ms   | 45.88ms   | 102.00ms   | 35.00ms   | 20.00ms    | 20.00ms   | 275.00ms   | 118.00ms  |
| /api/trpc/viewer.locationOptions                | 443.50ms   | 251.00ms  | 636.00ms   | 251.00ms  | 251.00ms   | 251.00ms  | 636.00ms   | 251.00ms  |
| /api/trpc/viewer.me                             | 353.63ms   | 106.50ms  | 174.00ms   | 98.00ms   | 48.00ms    | 48.00ms   | 912.00ms   | 174.00ms  |
| /api/trpc/viewer.public.i18n                    | 280.00ms   | 141.67ms  | 210.00ms   | 130.00ms  | 85.00ms    | 85.00ms   | 757.00ms   | 210.00ms  |
| /api/trpc/viewer.public.session                 | 383.00ms   | 214.00ms  | 218.00ms   | 218.00ms  | 210.00ms   | 210.00ms  | 721.00ms   | 218.00ms  |
| /api/trpc/viewer.teams.getUpgradeable           | 358.12ms   | 101.67ms  | 160.00ms   | 83.00ms   | 50.00ms    | 50.00ms   | 773.00ms   | 160.00ms  |
| /api/trpc/viewer.teams.hasTeamPlan              | 547.50ms   | 82.00ms   | 1013.00ms  | 82.00ms   | 82.00ms    | 82.00ms   | 1013.00ms  | 82.00ms   |
| /api/trpc/viewer.teams.list                     | 310.00ms   | 249.50ms  | 383.00ms   | 383.00ms  | 116.00ms   | 116.00ms  | 500.00ms   | 383.00ms  |
| /api/trpc/viewer.teams.listInvites              | 378.82ms   | 129.56ms  | 233.00ms   | 119.00ms  | 64.00ms    | 64.00ms   | 781.00ms   | 233.00ms  |
| /api/trpc/viewer.teams.listMembers              | 381.00ms   | 269.00ms  | 493.00ms   | 269.00ms  | 269.00ms   | 269.00ms  | 493.00ms   | 269.00ms  |
| /api/trpc/viewer.workflows.getByViewer          | 151.50ms   | 67.00ms   | 236.00ms   | 67.00ms   | 67.00ms    | 67.00ms   | 236.00ms   | 67.00ms   |
| /api/trpc/viewer.workflows.list                 | 213.25ms   | 181.00ms  | 255.00ms   | 289.00ms  | 73.00ms    | 73.00ms   | 289.00ms   | 289.00ms  |
